### ISSUE TITLE
[FEATURE] License Level parsing refactor

### ISSUE THREAD

**Issue by aritsune** [massif-press/compcon#628]
**Is your feature request related to a problem? Please describe.**
Right now, the only way to make a piece of licensed equipment available without a license is to put it under the GMS manufacturer. This tightly couples the app to the book data. Also, frames for a license are always fixed at rank 2 - in fact, a FRAME is *required* for a license to exist. These two things limit what homebrew (or even future official) content packs can do (GMS equipment that requires licensing, licenses that unlock a FRAME at rank 1 or 3, etc...), and can be hard to parse in code.

**Describe the solution you'd like**
A more standard way to assign a piece of equipment as "global" - probably setting `license_level` to `0`. Also, have licenses read the `license_level` value of the FRAME rather than ignore it and set it as 2 (though maybe keep 2 as a default fallback?).
Also, this is more ambitious and more likely to break, but we could possibly decouple licenses from FRAMEs altogether, and have them instead gather every item with the same `source`. This would allow for FRAME-less licenses. 

**Additional context**
This *will* require backwards-incompatible changes in lancer-data (& possibly other content packs, although if we simply set some defaults this might not be a problem.)

**Comment by jarena3** [comment]
removing this from 2.1. I think this makes the most sense to approach as we work on expanded homebrew functionality (2.2?)

**Comment by jarena3** [comment]
This should be working as expected as of 2.2.11
